master/internal/core.go

```diff
@@ -1142,16 +1142,17 @@ func (m *Master) Run(ctx context.Context, gRPCLogInitDone chan struct{}) error {
 	}
 
 	if isBrandNewCluster {
-		if password := m.config.Security.InitialUserPassword; password == "" {
-			log.Warn("This cluster was deployed without a default password for the built-in `determined` " +
-				"and `admin` users. You should set one using `det user change-password`. New clusters can be " +
-				"deployed with default passwords set using the `security.initial_user_password` setting.")
-		} else {
-			for _, username := range user.BuiltInUsers {
-				err := user.SetUserPassword(ctx, username, password)
-				if err != nil {
-					return fmt.Errorf("could not update default user password: %w", err)
-				}
+		password := m.config.Security.InitialUserPassword
+		if password == "" {
+			log.Error("This cluster was deployed without a default password for the built-in `determined` " +
+				"and `admin` users. New clusters can be deployed with default passwords set using the " +
+				"`security.initial_user_password` setting.")
+			return errors.New("could not deploy without default password")
+		}
+		for _, username := range user.BuiltInUsers {
+			err := user.SetUserPassword(ctx, username, password)
+			if err != nil {
+				return fmt.Errorf("could not update default user password: %w", err)
```

Comment:
Once more with the "default v/s initial" comment, but also, any feeling on also indicating which user failed in this string?

Reply:
Might be helpful; I'll at least spend a couple extra minutes thinking about what way of phrasing this error message could make it clearer what happened (it's probably going to be a DB connection timeout or similar kind of thing.)